src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +294,98 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
+        SnapshotModel snapshot = dataRepoFixtures.getSnapshot(steward(), snapshotSummary.getId());
+        assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
+        assertEquals("There should be 5 snapshot relationships", snapshot.getRelationships().size(), 5);
+
+        // fetch ACLs
+        logger.info("---- Dataset ACLs after snapshot create-----");
+        int datasetPlusSnapshotCount = retryACLUpdate(datasetName, datasetAclCount, ACLCheck.GREATERTHAN);
+        assertThat("There should be more ACLs on the dataset after snapshot create",
+            datasetPlusSnapshotCount, greaterThan(datasetAclCount));
+
+        //-----------delete snapshot------------
+        dataRepoFixtures.deleteSnapshot(steward(), snapshotSummary.getId());
+        logger.info("---- Dataset ACLs after snapshot delete-----");
+        int datasetMinusSnapshotACLCount = retryACLUpdate(datasetName, datasetAclCount, ACLCheck.EQUALTO);
+        assertEquals("We should be back to the same number of ACLs on the dataset after snapshot delete",
+            datasetAclCount, datasetMinusSnapshotACLCount);
+    }
+
+    private int fetchSourceDatasetAcls(String datasetName) throws Exception {
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+        bigQuery = BigQueryFixtures.getBigQuery(dataset.getDataProject(), stewardToken);
+
+        // Fetch BQ Dataset
+        String bqDatasetName = bigQueryPdao.prefixName(datasetName);
+        Dataset bq_dataset = bigQuery.getDataset(bqDatasetName);
+
+        // fetch ACLs
+        List<Acl> acls = bq_dataset.getAcl();
+        int aclCount = acls.size();
+
+        acls.forEach(acl ->  logger.info("Acl: {}", acl));
+        return aclCount;
+    }
+
+    enum ACLCheck {
+        GREATERTHAN,
+        EQUALTO
+    }
+
+    private int retryACLUpdate(String datasetName, int datasetAclCount, ACLCheck check) throws Exception {
+        double maxDelayInSeconds = 60;
+        int wait_interval;
+        // Google claims it can take up to 7 minutes to update ACLs
+        int sevenMinutePlusBuffer = (7 * 60) + 20;
+        int datasetPlusSnapshotCount = 0;
+        int totalWaitTime = 0;
+        double n = 1;
+        while (totalWaitTime < sevenMinutePlusBuffer) {
+            datasetPlusSnapshotCount = fetchSourceDatasetAcls(datasetName);
+            if (check.equals(ACLCheck.GREATERTHAN)) {
+                if (datasetPlusSnapshotCount > datasetAclCount) {
+                    break;
+                }
+            } else if (check.equals(ACLCheck.EQUALTO)) {
+                if (datasetPlusSnapshotCount == datasetAclCount) {
+                    break;
+                }
+            } else {
+                logger.error("Not a valid enum value for ACLCheck");
+            }
+            double delayInSeconds =  ((1d / 2d) * (Math.pow(2d, n) - 1d));
+            wait_interval = maxDelayInSeconds < delayInSeconds ?
```

Comment:
Couple things:
- The variable should use camel case (e.g. `waitInterval`)
- This is easier to read as `Math.min(maxDelayInSeconds, delayInSeconds)`